universal-login-web3/src/ui/react/Confirmation/TransactionSpeed.tsx

```diff
@@ -1,29 +1,33 @@
 import React from 'react';
 import styled from 'styled-components';
+import {GasMode} from '@universal-login/commons';
 
-export interface TransactionSpeedProps {
+export interface TransactionSpeedSelectionProps {
   selectedValue: string;
   onChange: (value: string) => void;
 }
 
-export const TransactionSpeed = ({selectedValue, onChange}: TransactionSpeedProps) => (
-  <Row>
-    <RadioButton
-      value="cheap"
-      selectedValue={selectedValue}
-      onChange={onChange}
-      time="1 day"
-    />
-    <RadioButton
-      value="fast"
-      selectedValue={selectedValue}
-      onChange={onChange}
-      time="1 min"
-    />
-  </Row>
-);
+export interface TransactionSpeedProps extends TransactionSpeedSelectionProps {
+  gasModes: GasMode[];
+}
+
+export const TransactionSpeed = ({gasModes, selectedValue, onChange}: TransactionSpeedProps) => {
+  return (
+    <Row>
+      {gasModes.map(({name, usdAmount}) => (
+        <RadioButton
+          key={name}
+          value={name}
+          selectedValue={selectedValue}
+          onChange={onChange}
+          time="1 day"
```

Comment:
Is 1 day correct for each gasMode?

Reply:
It is mock up, so i just write value. We don't have any service which calculate it. Which value do you suggest write here?